rest-service/manager_rest/storage/resource_models.py

```diff
@@ -590,9 +590,22 @@ def _is_workflow_available(self, workflow):
             return True
         if not rules.get('available', True):
```

Comment:
Perhaps the entire rules thing should be done by matching each specified key against a dict that defines the function used to validate them?

e.g. return all(MAPPING[rules[rule]] for rule in rules if rule in MAPPING)

Reply:
Yeah, maybe.  I'd like to work on it once I implement some other rules and we'll see some pattern(s)

Reply:
It might be worth doing that now (see my response to your other comment below) to make that be the pattern, and only break it out into something more bespoke if we discover that we really can't fit the pattern- though presumably all of the rules will have an accompanying value from which can be determined whether that rule considers the workflow to be available or not, so this pattern should hold.

Reply:
I think it's not going to be that simple.  See @tehasdf 's comment on "AND-ing" `node_instances_active` rule

Reply:
This will AND them- using all(<conditions>)